src/WooCommerce.php

```diff
@@ -389,14 +397,25 @@ public static function woocommerce_save_product_variation($post_id, $loop) {
     foreach ($custom_fields as $field) {
       if (isset($_POST[$field]) && isset($_POST[$field][$loop])) {
         if ($_POST[$field][$loop]) {
-          update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          if ($field !== '_' . Plugin::PREFIX . '_gtin') {
+            update_post_meta($post_id, $field, $_POST[$field]);
+          } else if (!self::is_existing_gtin($variation_id, $_POST[$field][$loop])) {
+            // GTIN is only saved if unique.
+            update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          } else {
+            $gtin_error = true;
+          }
         }
         else {
           delete_post_meta($variation_id, $field);
         }
       }
     }
 
+    if ($gtin_error) {
+      \WC_Admin_Meta_Boxes::add_error(__('At least one entered GTIN in product variations isn\'t unique and couldn\'t be saved. Please enter unique GTINs and save again. All other changes have been saved.', Plugin::L10N));
+    }
```

Comment:
Can not understand why we are displaying the error twice with a different message. 

I mean, I think we could extract the error message into a function and call it from `woocommerce_process_product_meta ` and `woocommerce_save_product_variation ` but using always the same message. 

If do so, please ensure to clean as well the related translation.